packages/analytics-browser/src/storage/local-storage.ts

```diff
@@ -1,8 +1,26 @@
 import { getGlobalScope } from '@amplitude/analytics-client-common';
 import { BrowserStorage } from './browser-storage';
+import { Logger } from '@amplitude/analytics-types';
 
+interface LocalStorageOptions {
+  logger?: Logger;
+}
 export class LocalStorage<T> extends BrowserStorage<T> {
-  constructor() {
+  logger?: Logger;
+
+  constructor(config?: LocalStorageOptions) {
     super(getGlobalScope()?.localStorage);
+    this.logger = config ? config.logger : undefined;
+  }
+
+  async set(key: string, value: T): Promise<void> {
+    if (Array.isArray(value) && value.length > 1000) {
+      const droppedEventsCount = value.length - 1000;
+      value.slice(droppedEventsCount);
+      const errorMessage = `Dropped ${droppedEventsCount} events because the queue length exceeded 1000.`;
+      this.logger ? this.logger.error(errorMessage) : console.error(errorMessage);
```

Comment:
We should avoid console.log() to honor logLevel.

```suggestion
      this.logger?.error(errorMessage);
```

Reply:
So it will drop without any logs.